### PR TITLE
fix(useElementSize): use contentBoxSize as an array in useElementSize

### LINKED ISSUE #2519: useElementSize has a bug in fireFox version 80
### Describe the bug

ResizeObserverEntry is an object in fireFox version 80, but it is used as an array

### Reproduction

https://play.vueuse.org/#N4IgDghgxg1hDmBTAziAXAbVAOwgW0XRADcBXRAWgBNE8BLEAGhGQHtSAnKQtEU7MDHhMQNZFA50wAFzqtsRAGrkABABFadFQAoAFhAA2AMxV1sKgGIdE2KLoCUp5CogqaxRAdZgz8FaVkDOmkATxcDLwB3ZxD2FWlWFUjJaUQVAFVsOg8OZEMVZTSAGToAIw4ISRQVI1YOAtUAJhUAMhUAZhFOAyJdaWkwZDQAemH+QXgAOihWPGGyShp6YaDS4bMaAA9JvAArVABfRhx8HhAAAQXSZERh5H1rKhE2Tm4icaERMQkpWXkiADKD0QVAaiHSN38gWCdBQky6HB6vD6AyGow+Uxmc0u5Gut3ulRB5wAnJMAAzk9bYLY7fYgI4nAhEHGIPHDGbWZ7sLhnDFfFA-GRyBS8ADCrAiiCgf3MrBMKBu2Fk+UKKnFeDArGQMPkGWhshQCKRIBRgxGYwEQmms2GLLZHMQJPJlI2iG2e0OAF1mEY6AZDWhQABBMBgSYLdCgVIagwQVJEAA8qU20kJrmsRgAvAAdECeXMqKCx5DIHMgazagBeiA4BeIFGT0jLjYLwwAfNnsMBgFXEAcDs9BdIiHQNXVpCpgCprNBZB5GNPECYDjUOLMVAByBYbzujzUcCdTvEAUX9BCVALo1ZUK6Ma7wm7tN3ZdUQO9s8mQE88Kkzi6M2jYKQET2J2MzYF+Ki9r+i6ztkiDaJ2Kj+Dcp60DY0iXtWiHmMhKieIwSF4VOkR0FQ0i6GgKhkguuiIHQ8B9FRZI3oRuHIVOpSsJsVEblxHA0BwFBcZsG6sUR9hsfY9LMOKdRUGYcaIOG5CRiA0ZgLG8a8AmCnEERtRKmWeDyKwuZEaU8BlkEjHSBQACsZIsVQlQwGgLkcDADlOeZuFGP6mxlgAfoWEq+chjZltwSo1uFKhgGWjRxfxgllmu-A0FQvkdrhCbIJA5iRbmAAsmwGLmbbdioxCGKo-YJncBU5cheUFfEbpNrmyAPh5bk2X0FDEk5biuWg3jQMEITeQWeDNhQSUgG2ACy7CQlVsalD+9WNRA2A5Q1eltoOkgyEQNC+tgiAAAproM2jAERG2eFRALSJI2DwGxyE1QY5BUQAcqQeCbRwbEHNJ-YHEAA

image

image

### System Info

```Shell
System:
    OS: Windows 10 10.0.19043
    CPU: (16) x64 Intel(R) Core(TM) i7-10700 CPU @ 2.90GHz  
    Memory: 13.06 GB / 31.72 GB
  Binaries:
    Node: 16.16.0 - D:\安装包\node-v16.16.0-win-x64\node.EXE
    Yarn: 1.12.3 - C:\Program Files (x86)\Yarn\bin\yarn.CMD 
    npm: 8.11.0 - D:\安装包\node-v16.16.0-win-x64\npm.CMD   
  Browsers:
    Edge: Spartan (44.19041.1023.0), Chromium (92.0.902.55) 
    Internet Explorer: 11.0.19041.1
  npmPackages:
    @vueuse/core: 9.3.1 => 9.3.1
    @vueuse/head: ^0.7.4 => 0.7.4
    vue: ^3.2.37 => 3.2.37
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/vueuse/vueuse/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a VueUse issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/core instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vueuse/vueuse/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by threedayAAAAA**
### Description

Fixes https://github.com/vueuse/vueuse/issues/2519

### Additional context

---

### What is the purpose of this pull request? 

- [x] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/core/useElementSize/index.ts [hunk 1]
```diff
@@ -35,9 +35,10 @@ export function useElementSize(
           ? entry.contentBoxSize
           : entry.devicePixelContentBoxSize
 
-      if (boxSize) {
-        width.value = boxSize.reduce((acc, { inlineSize }) => acc + inlineSize, 0)
-        height.value = boxSize.reduce((acc, { blockSize }) => acc + blockSize, 0)
+      const formatBoxSize = Array.isArray(boxSize) ? boxSize : [boxSize]
+      if (formatBoxSize) {
```
Comment (vaakian) [inline packages/core/useElementSize/index.ts:39]: `boxSize` may be `undefined`, so formatBoxSize may goes `[ undefined ]` and go into the if block. 
Put the check inside if block would be better.

Comment (threedayAAAAA) [reply packages/core/useElementSize/index.ts:39]: Is a good suggestion

Comment (threedayAAAAA) [reply packages/core/useElementSize/index.ts]: I have fixed it, please review again @vaakian

**Review by vaakian — DISMISSED** [review]

**Comment by threedayAAAAA** [comment]
Please merge this pr @antfu

**Review by dpschen — DISMISSED** [review]

**Comment by MichalPniak** [comment]
Hi, when can we expect a bugfix release?

**Pull request merged**